### DISCUSSION TITLE
Why are multiline string so restrictive with mandating """\n at start?

### DISCUSSION THREAD

**Discussion by eugenesvk** [kdl-org/kdl#455] [General]
I get that the last `"""` must only have whitespace to signal the amount of dedenting that should happen (though this also seems questionable, you could get that signal from just the last line?), but then what's up with the first `"""` that **must** be followed by newline right away, thus eating up more vertical space than seems necesssary?

**Comment by eilvelia** [comment]
Isn't
```
"""  foo
  bar
  text
"""
```
(or other possible variations)

a bit weird? That would go against the intuition that the multiline string contents are what they look in the source text, but without taking into account whitespace indentation.

FWIW this rule is shared by many other languages

**Reply by eugenesvk** [reply]
```
"Sure, for short tables"="""you'd prefer a newline. But then for long test this isn't that important,
    - you just want to be able to add an indented example
  or just have a more readable text without soft wrapping"""
```
And the dedent whitespace count could come from the last line, so I'm not even sure you need to mandate a separate line for the ending `"""`.

So with the optional newlines you'd only add them if you need that extra visual fix

**Comment by zkat** [comment]
To prevent unfortunate mix-ups and make parsing and other sorts of automated detection easier. They are, essentially, two completely different types of data, and being able to quickly disambiguate between them is helpful.

Same reason we even use triple quotes.
